apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolKey.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Google.Api.Gax.Grpc;
+using Google.Apis.Auth.OAuth2;
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// Holds a key that we use to look up cached sessions.
+    /// </summary>
+    internal struct SessionPoolKey
```

Comment:
I'd strongly advise either making this immutable, and potentially a class unless you have a reason not to. If it's a key for lookup purposes, let's implement `IEquatable<SessionPoolKey>` either way...